connect/src/spirc.rs

```diff
@@ -123,6 +152,7 @@ pub enum SpircCommand {
     VolumeDown,
     Shutdown,
     Shuffle,
+    SetSessionEventCallback(Option<SessionEventCallback>),
```

Comment:
For whatever reason I don't understand passing the callback in does not work. As far as if it should be in the channel I can do that if that's what everyone wants I don't really care I did it this way because it was the only way I could figure out how to make it work. I used the sink event callback as an example.

Reply:
What does it say when you try to pass it as a parameter? I think in `main` there are a couple of other instances where we pass functions into constructors, does any of that work for you?

Would be nice if it were a channel but don't break your head over it. Still would be nice if we could get rid of this "dirty" enum variant though.

Reply:
I didn't try to get it in the channel I imagine that wouldn't be too hard I can do that tomorrow after work.